secret/secret_test.go

```diff
@@ -120,58 +119,52 @@ func (m *MockSecrets) RecordSSLCertInfo(configVar *model.ConfigurationVariable)
 	m.Called(configVar)
 }
 
-func (m *MockSecrets) GenerateSSLCerts(secrets, updates *v1.Secret) (dirty bool) {
-	results := m.Called(secrets, updates)
-	return results.Bool(0)
+func (m *MockSecrets) GenerateSSLCerts(secrets, updates *v1.Secret) {
+	m.Called(secrets, updates)
 }
 
 func TestUpdateSecretsWhenCreatingOrUpdating(t *testing.T) {
 	t.Parallel()
 
 	t.Run("Neither Create Nor Update called", func(t *testing.T) {
 		var s MockSecretInterface
-		s.On("Create", (*v1.Secret)(nil)).Return(nil, nil)
-		s.On("Update", (*v1.Secret)(nil)).Return(nil, nil)
+		secrets := &v1.Secret{Data: map[string][]byte{}}
 
-		UpdateSecrets(&s, nil, false, false)
-		s.AssertNotCalled(t, "Create", nil)
-		s.AssertNotCalled(t, "Update", nil)
+		s.On("Create", secrets).Return(nil, nil)
+		s.On("Update", secrets).Return(nil, nil)
+
+		UpdateSecrets(&s, secrets, false)
+		s.AssertNotCalled(t, "Create", secrets)
+		s.AssertNotCalled(t, "Update", secrets)
 	})
 
 	t.Run("Create, but don't update", func(t *testing.T) {
 		var s MockSecretInterface
-		s.On("Create", (*v1.Secret)(nil)).Return(nil, nil)
-		s.On("Update", (*v1.Secret)(nil)).Return(nil, nil)
-
-		UpdateSecrets(&s, nil, true, false)
-		s.AssertCalled(t, "Create", (*v1.Secret)(nil))
-		s.AssertNotCalled(t, "Update", nil)
-	})
+		secrets := &v1.Secret{Data: map[string][]byte{}}
 
-	t.Run("Create and update called", func(t *testing.T) {
-		var s MockSecretInterface
-		s.On("Create", (*v1.Secret)(nil)).Return(nil, nil)
-		s.On("Update", (*v1.Secret)(nil)).Return(nil, nil)
+		s.On("Create", secrets).Return(nil, nil)
+		s.On("Update", secrets).Return(nil, nil)
 
-		UpdateSecrets(&s, nil, true, true)
-		s.AssertCalled(t, "Create", (*v1.Secret)(nil))
-		s.AssertNotCalled(t, "Update", (*v1.Secret)(nil))
+		UpdateSecrets(&s, secrets, true)
+		s.AssertCalled(t, "Create", secrets)
+		s.AssertNotCalled(t, "Update", secrets)
```

Comment:
Assert that it only has the expected secrets in it? (I.e. that it's not dirty)

Same for the next one.